tests/components/media_player/test_async_helpers.py

```diff
@@ -139,121 +138,89 @@ async def async_toggle(self):
-        ).result()
-        assert self.player.volume_level == 0.3
-
-    def test_media_play_pause(self):
-        """Test the media_play_pause helper function."""
-        assert self.player.state == STATE_OFF
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_media_play_pause(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_PLAYING
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_media_play_pause(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_PAUSED
-
-    def test_toggle(self):
-        """Test the toggle helper function."""
-        assert self.player.state == STATE_OFF
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_toggle(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_ON
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_toggle(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_OFF
+@pytest.fixture
+def async_player(hass):
+    """Fixture of an asynchronous media player."""
+    return AsyncMediaPlayer(hass)
+
+
+@pytest.fixture
+def sync_player(hass):
+    """Fixture of a synchronous media player."""
+    return SyncMediaPlayer(hass)
+
+
+async def test_async_volume_up(hass, async_player):
+    """Test the volume_up helper function."""
+    assert async_player.volume_level == 0
+    await asyncio.ensure_future(
+        async_player.async_set_volume_level(0.5), loop=hass.loop
+    )
+    assert async_player.volume_level == 0.5
+    await asyncio.ensure_future(async_player.async_volume_up())
+    assert async_player.volume_level == 0.6
+
+
+async def test_async_volume_down(hass, async_player):
+    """Test the volume_down helper function."""
+    assert async_player.volume_level == 0
+    await asyncio.ensure_future(
+        async_player.async_set_volume_level(0.5), loop=hass.loop
+    )
+    assert async_player.volume_level == 0.5
+    await asyncio.ensure_future(async_player.async_volume_down(), loop=hass.loop)
+    assert async_player.volume_level == 0.4
+
+
+async def test_async_media_play_pause(hass, async_player):
+    """Test the media_play_pause helper function."""
+    assert async_player.state == STATE_OFF
+    await asyncio.ensure_future(async_player.async_media_play_pause(), loop=hass.loop)
+    assert async_player.state == STATE_PLAYING
+    await asyncio.ensure_future(async_player.async_media_play_pause(), loop=hass.loop)
+    assert async_player.state == STATE_PAUSED
+
+
+async def test_async_toggle(hass, async_player):
+    """Test the toggle helper function."""
+    assert async_player.state == STATE_OFF
+    await asyncio.ensure_future(async_player.async_toggle(), loop=hass.loop)
+    assert async_player.state == STATE_ON
+    await asyncio.ensure_future(async_player.async_toggle(), loop=hass.loop)
+    assert async_player.state == STATE_OFF
```

Comment:
The tests for the async media player can be removed.
```suggestion
```

Reply:
So I kept the tests for `async_toggle`, `async_volume_up`, `async_volume_down` and `async_media_play_pause` and deleted a lot of the other stuff. I took a look at the parent class and the tests makes more sense to me now. I will make an effort to update!